lib/MySQL_Session.cpp

```diff
@@ -4755,6 +4788,17 @@ void MySQL_Session::handler___status_CONNECTING_CLIENT___STATE_SERVER_HANDSHAKE(
 		client_myds->rbio_ssl = BIO_new(BIO_s_mem());
 		client_myds->wbio_ssl = BIO_new(BIO_s_mem());
 		client_myds->ssl=SSL_new(GloVars.global.ssl_ctx);
+		// get the current 'tls_version' specified in the global variable 'admin-tls_version'
+		std::string tls_version_var = GloAdmin->get_variable(const_cast<char*>("tls_version"));
```

Comment:
A MySQL worker thread (in `MySQL_Thread`, `MySQL_Session`, or otherwise), should never call `GloAdmin->get_variable()` without getting a global lock on `GloAdmin` itself, otherwise dirty reads can happen.
In fact, a MySQL worker thread should never access admin variables.
This should be implemented as a `mysql-` , `mysql-tls_version`, and follow the same pattern of other mysql variable: a global variable in `MySQL_Threads_Handler`, and each thread should have its own local storage copy in `mysql_thread___tls_version` that can access without locking.

Reply:
I thought about this, but since the original request specifically asked for an Admin variable I went that way. I also thought about the dirty reads, but since this is something that happens per connection, didn't wanted to introduce a lock there. Thanks for the feedback, changing it in favor of 'mysql-tls_version'.